src/main/kotlin/com/terraformation/backend/seedbank/model/Accession.kt

```diff
@@ -151,6 +208,35 @@ data class AccessionModel(
     }
   }
 
+  private fun getManualStateTransition(
+      newModel: AccessionModel,
+      clock: Clock
+  ): AccessionStateTransition? {
+    val seedsRemaining = newModel.calculateRemaining(clock)
+    val allSeedsWithdrawn = seedsRemaining != null && seedsRemaining.quantity <= BigDecimal.ZERO
+    val oldState = state ?: AccessionState.AwaitingProcessing
+    val newState = newModel.state ?: AccessionState.AwaitingProcessing
+    val alreadyCheckedIn = oldState != AccessionState.AwaitingCheckIn
+    val revertingToAwaitingCheckIn =
+        alreadyCheckedIn && newModel.state == AccessionState.AwaitingCheckIn
+    val addingSeedsWhenUsedUp =
```

Comment:
Just a weird corner case since the client supports saving accessions without a quantity set, if seedsRemaining is null.. we may flag this as addingSeedsWhenUsedUp - can only happen from incorrect usage of client so maybe it's ok.

Reply:
I can't seem to figure out that corner case -- can you describe the sequence of changes that would end up going wrong? I definitely want this to be as airtight as possible.

Reply:
Saving an accession with no quantity set and updating state as 'used up'.
A subsequent update to this accession would move its state to 'in storage' (addingSeedsWhenUsedUp will be true).
This is a nonsensical corner case as it makes no sense but client would support saving such an accession with no quantity. I could be wrong about the 'used up' state being supported for such an accession.

Reply:
Ah, got it. Yeah, that'd end up wrong. The real bug there is you shouldn't be able to set the state to `UsedUp` if there's no quantity. Just updated the PR with a fix for that (and a test case to cover it).

Reply:
Thanks @sgrimm , changes look good. I'll create a ticket for FE/UX to handle this case as well.